crates/findex/src/implementations/redis/findex.rs

```diff
@@ -0,0 +1,434 @@
+                    Uid::from(uid)
+                })
+                .collect(),
+        ))
+    }
+
+    async fn fetch_entry_table(
+        &self,
+        entry_table_uids: Uids<UID_LENGTH>,
+    ) -> Result<EncryptedMultiTable<UID_LENGTH>, FindexRedisError> {
+        trace!(
+            "fetch_entry_table num keywords: {}:",
+            entry_table_uids.0.len(),
+        );
+        // guard against empty uids
+        if entry_table_uids.0.is_empty() {
+            return Ok(EncryptedMultiTable::default());
+        }
+
+        // build Redis keys
+        let keys: Vec<Vec<u8>> = entry_table_uids
+            .0
+            .iter()
+            .map(|uid| key(FindexTable::Entry, uid))
+            .collect();
+
+        // mget the values from the Redis keys
+        let values: Vec<Vec<u8>> = self.mgr.clone().mget(keys).await?;
+
+        // discard empty values
+        let tuples = entry_table_uids
+            .0
+            .into_iter()
+            .zip(values)
+            .filter(|(_uid, v)| !v.is_empty())
+            .collect::<Vec<_>>();
+        trace!("fetch_entry_table non empty tuples len: {}", tuples.len(),);
+
+        Ok(EncryptedMultiTable(tuples))
+    }
+
+    async fn fetch_chain_table(
+        &self,
+        chain_table_uids: Uids<UID_LENGTH>,
+    ) -> Result<EncryptedTable<UID_LENGTH>, FindexRedisError> {
+        trace!(
+            "fetch_chain_table num entries: {}:",
+            chain_table_uids.0.len(),
+        );
+
+        //guard against empty uids
+        if chain_table_uids.0.is_empty() {
+            return Ok(EncryptedTable::default());
+        }
+
+        let keys: Vec<Vec<u8>> = chain_table_uids
+            .0
+            .iter()
+            .map(|uid| key(FindexTable::Chain, uid))
+            .collect();
+        let values: Vec<Vec<u8>> = self.mgr.clone().mget(keys).await?;
+
+        Ok(EncryptedTable::from(
+            chain_table_uids
+                .0
+                .into_iter()
+                .zip(values)
+                .filter(|(_uid, v)| !v.is_empty())
+                .collect::<HashMap<Uid<UID_LENGTH>, Vec<u8>>>(),
+        ))
+    }
+
+    async fn upsert_entry_table(
+        &self,
+        modifications: UpsertData<UID_LENGTH>,
+    ) -> Result<EncryptedTable<UID_LENGTH>, FindexRedisError> {
+        // Prevent upserting while compacting
+        if self.compact_lock.try_read().is_err() {
+            return Err(FindexRedisError::Compacting);
+        }
+
+        trace!("upsert_entry_table num keywords {:?}", modifications.len());
+
+        let mut rejected = EncryptedTable::default();
+        for (uid, (old_value, new_value)) in modifications {
+            let value: Vec<u8> = self
+                .upsert_script
+                .arg(key(FindexTable::Entry, &uid))
+                .arg(old_value.unwrap_or_default())
+                .arg(new_value)
+                .invoke_async(&mut self.mgr.clone())
```

Comment:
`self.mgr` seems to be a non-trivial object. Have you considered using an `Arc` instead of cloning it around?

Reply:
This is the doc recommended usage